cpp/include/cudf/aggregation.hpp

```diff
@@ -286,6 +287,40 @@ std::unique_ptr<Base> make_nth_element_aggregation(
 template <typename Base = aggregation>
 std::unique_ptr<Base> make_row_number_aggregation();
 
+/**
+ * @brief Factory to create a EWMA aggregation
+ * `EWMA` returns a non-nullable column with the same type as the input,
+ * whose values are the exponentially weighted moving average of the input
+ * sequence. Let these values be known as the y_i.
+ *
+ * EWMA aggregations are parameterized by a center of mass (`com`) which
+ * affects the contribution of the previous values (y_{i-1} ... y_0) in
+ * computing the y_i.
+ *
+ * EWMA aggregations are also parameterized by a boolean `adjust`.
+ * Special considerations have to be given to the mathematical treatment of
+ * the first value of the input sequence. There are two approaches to this,
+ * one which considers the first value of the sequence to be the expoential
+ * weighted moving average of some infinite history of data, and one which
+ * takes the first value to be the only datapoint known. These assumptions
+ * lead to two different formulas for the y_i. `adjust` selects which one.
```

Comment:
while two values are adequate to communicate which formula to use, I think it would be prudent to use an enum class here with relevant names for each enum value. Not sure what the names would be, though...

https://github.com/pandas-dev/pandas/issues/8861

Reply:
👍 I think I remember it saying something about this in the developer guide as well. How about `INFINITE_HISTORY` and `FINITE_HISTORY`?

Reply:
What would you name the enum itself? I don't particularly like
```
enum class adjust {
  FINITE_HISTORY,
  INFINITE_HISTORY
};
```
The name `adjust` doesn't really tell you anything about what it means. Maybe
```
enum class HISTORY {
  FINITE,
  INFINITE
}
```
I can't think of any other options here. Even if further levers are added to the EWM calculation I think they'll be orthogonal to this one, so this naming suggestion doesn't feel that constraining to me.